tests/integration/conftest.py

```diff
@@ -38,12 +39,36 @@ def charm_base(request) -> str:
 @pytest.fixture(scope="module")
 async def slurmd_charm(ops_test: OpsTest):
     """Build slurmd charm to use for integration tests."""
-    charm = await ops_test.build_charm(".")
-    return charm
+    return await ops_test.build_charm(".")
+
+
+@pytest.fixture(scope="module")
+async def slurmctld_charm(ops_test: OpsTest):
+    """Build slurmctld charm to use for integration tests.
+
+    Notes:
+        Returns string "slurmctld" if ../slurmrestd-operator does not exist.
+    """
+    if (slurmctld := PARENT_DIR / "slurmctld-operator").exists():
+        return await ops_test.build_charm(slurmctld)
```

Comment:
We should make the integration test enhancements a separate pull request if we want to add a custom flag for using local charms versus what is currently published in the charm edge channels. I mostly just added this mechanism since slurmd would have issues marking the compute node as active with the edge version of the slurmctld operator since slurmctld was expecting etcd to be running/updated.

Reply:
Local charm builds removed in commit https://github.com/omnivector-solutions/slurmd-operator/pull/21/commits/a13f712e89fd527de9afdc9f4e27bb93908387d3